### PR TITLE
Aluminum and copper utility expansions

### PR DESCRIPTION
**Pull request by DangerNoodle**
Part one of (possibly) two for the Mining Mod foundation, making the native metals more usable by the player. Current focus is aluminum, and copper to a lesser extent. See issue: https://github.com/CleverRaven/Cataclysm-DDA/issues/18478

Added an aluminum scrap item, as a proper single-volume aluminum material for crafting and repair.

Antenna, directional antenna, and metal tongs have all been changed to allow aluminum as an optional material. Extra-light frames have been changed to use aluminum exclusively.

Added aluminum scraps to list of materials usable for aluminum powder, and rebalanced amount of cans/foil required to retain a sane ratio between scrap, foil, and cans.

Added recipe for making aluminum foil, and added it as a component for certain food recipes. Vegetables are the most obvious example of food that is typically wrapped in foil if cooked at a campfire, so most cooked vegetable recipes now use either boiling, baking, or frying.

Pies and pizzas also made to require foil, requiring painstaking formatting long lists of veggies and fruit ingredients.

Regarding copper, made copper wire actually use copper to make, electronics-wise that was the only major missing function.

### CODE REVIEW

**Comment by DangerNoodle** [comment]
Apologies for this, but will check up on this in the morning, if I have free time. Seriously in need of sleep, and having to be up early.

**Comment by Coolthulhu** [comment]
> Added an aluminum scrap item

You should modify the aluminum ingot instead. No need to have both.

> Added recipe for making aluminum foil

Pretty sure making foil isn't all that simple (if at all doable) without industrial tools.

> added it as a component for certain food recipes

Would be fine if it was just an alternative component, but strictly requiring foil is not. Tools that are too specific end up annoying.

Also, the idea about crafting aluminum scrap from cans was brought up and dropped before because realism, as aluminum can't be smelted like that without oxidizing.

**Comment by mugling** [comment]
> Also, the idea about crafting aluminum scrap from cans was brought up and dropped before because realism, as aluminum can't be smelted like that without oxidizing.

Is it necessarily bad from a gameplay POV? That said aluminium being rare and requiring scavenging is probably more interesting than a tedious recipe for collecting hundreds of aluminum cans.

**Comment by Coolthulhu** [comment]
> Is it necessarily bad from a gameplay POV?

Most likely wouldn't result in any serious gain.
Aluminum would need good uses to warrant collecting enough cans to smelt them.
Especially if we allow those aluminum extra light frames, which would most likely be a better source.

**Comment by BorkBorkGoesTheCode** [comment]
I proposed the idea to get rid of the aluminum cans.

**Comment by DangerNoodle** [comment]
> You should modify the aluminum ingot instead. No need to have both.

Sounds like a saner idea, yes.

> Pretty sure making foil isn't all that simple (if at all doable) without industrial tools.

Not sure how to tackle that issue. It does add to the utility of aluminum though.

> Would be fine if it was just an alternative component, but strictly requiring foil is not. Tools that are too specific end up annoying.

Seeing as some cooked vegetable items require water and some do not, it might be sensible to add foil as an alternative to water. Either as a separate recipe, or by making it an alternative component? Not sure how to handle pies and pizza though. Those are specific, mid-level items that few players are going to be making except for fun, though. Unlike the basic cooked veggies, where not being able to make it might mean starving.

**Comment by DangerNoodle** [comment]
For now though, I am afraid I can not mess with it due to having to leave in a bit. Will attempt to resolve linting and current oddities when I am able to later.

**Comment by Zireael07** [comment]
What can be done IRL with aluminium ingots without industrial tools/oxidizing? @Coolthulhu? I recall this topic being discussed several times but no specifics beyond 'can't smelt cans' ...

**Comment by DangerNoodle** [comment]
Currently reworking most of the cooked vegetable items to allow aluminum foil as an alternative to water, mainly where the quality demands do not make it clear that the player is boiling

Regarding making aluminum foil, I am looking into the idea, and it might be possible to produce sheets using non-industrial methods after casting the aluminum into a sheet ingot, but the main loss is in uniformity of thickness, and tedium.

**Comment by DangerNoodle** [comment]
Short on time, so I will have to overview further lint failures and devise a saner recipe for aluminum foil later on.

File: data/json/recipes/other/materials.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "difficulty": 1,
     "time": 15000,
     "batch_time_factors": [ 90, 4 ],
-    "reversible": true,
+    "reversible":
+ true,
```
Comment (mugling) [inline data/json/recipes/other/materials.json]: This doesn't look right...

Comment (DangerNoodle) [inline data/json/recipes/other/materials.json]: I am questioning how I managed to do that.

Comment (DangerNoodle) [inline data/json/recipes/other/materials.json]: I must have accidentally hit enter in the process.

File: data/json/recipes/food/veggy.json [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+      [ ["apple", 2], ["apple_canned", 1], ["apple_vac", 1], ["apricot", 2], ["banana", 2], ["blackberries", 2], ["blueberries", 2], ["cherries", 2], ["coconut", 2], ["cranberries", 6], ["dry_fruit", 2], ["grapefruit", 2], ["grapes", 4], ["irradiated_apple", 2], ["irradiated_apricot", 2], ["irradiated_banana", 2], ["irradiated_blackberries", 2], ["irradiated_blueberries", 2], ["irradiated_cherries", 2], ["irradiated_cranberries", 6], ["irradiated_grapefruit", 2], ["irradiated_grapes", 4], ["irradiated_kiwi", 2], ["irradiated_lemon", 2], ["irradiated_mango", 2], ["irradiated_melon", 4], ["irradiated_orange", 2], ["irradiated_papaya", 2], ["irradiated_peach", 2], ["irradiated_pear", 2], ["irradiated_pineapple", 2], ["irradiated_plums", 2], ["irradiated_pomegranate", 2], ["irradiated_raspberries", 2], ["irradiated_rhubarb", 4], ["irradiated_strawberries", 2], ["irradiated_watermelon", 1], ["can_peach", 2], ["juice_pulp", 4], ["kiwi", 2], ["lemon", 2], ["mango", 2], ["melon", 4], ["orange", 2], ["papaya", 2], ["peach", 2], ["pear", 2], ["pineapple", 2], ["plums", 2], ["pomegranate", 2], ["raspberries", 2], ["rhubarb", 4], ["strawberries", 2], ["watermelon", 1] ],
```
Comment (DangerNoodle) [inline data/json/recipes/food/veggy.json]: This is a massive flaw in the current lint standards.

**Comment by DangerNoodle** [comment]
I would like to state that manually format-fixing the endless list of fruit usable in fruit pies is the biggest headache I have encountered thus far.

**Comment by DangerNoodle** [comment]
And the lint failures continue.

**Comment by chaosvolt** [comment]
Relevant to those ideas:

12:19 PM - Chaosvolt: You're officially the metal nerd here. How would you go about making aluminum foil without industrial tools? :V
12:19 PM - : hmm, a tamp and a lot of hammering
12:20 PM - : ooh, or if you had two anvils, you could line one up to drop on the other
12:20 PM - Chaosvolt: From an ingot, or casting it out into a roughly flat sheet first?
12:21 PM - : hmm, probably cast it flat first, aluminum can be brittle when cast thick
12:21 PM - : though you might have impurity issues if casting flat due to surface area
12:21 PM - : aluminum oxidizes stupid fast
12:22 PM - Chaosvolt: Ideally having some samples of sheet metal to use as a surface, and possibly a second to sandwich it?
12:23 PM - : samples of sheet metal as a surface? that is going to ruin the samples
12:24 PM - : foil is kind of difficult to make without rolling presses
12:24 PM - Chaosvolt: As I figured. ;w;
12:25 PM - : i mean you could make a rolling press though
12:25 PM - : if you are willing to sacrifice a car o3o
12:29 PM - Chaosvolt: I was thinking more along the lines of parts that are reasonably round, designed to rotate, and able to take a lot of weight.
12:30 PM - : you mean like a car o3o
12:31 PM - Chaosvolt: I was thinking more like take two wheels off, remove the tires...wait, you'd want something round and FLAT. -_-
12:31 PM - : yes, lathe down the wheels to a flat surface
12:32 PM - Chaosvolt: At that point it sounds like heating it up a bit short of melting point, sandwiching it between two flat objects, and dropping something heavy on it would be less hassle.
12:35 PM - : you wouldnt have a very large amount though
12:35 PM - : remember that rolling is used because it needs far less pressure
12:36 PM - Chaosvolt: True. :V
12:36 PM - : and you can make longer, bigger pieces
12:36 PM - Chaosvolt: I can see this being adequate for the derp game that provoked this question though. o3o
12:37 PM - : what, cannibalizing a car to make a rolling press?
12:37 PM - Chaosvolt: No. Someone's trying to figure out a sane recipe for making aluminum foil in Derp Days Ahead.
12:41 PM - : im just saying that you can make a LOT of industrial machines with a car and a bit of creativity
12:42 PM - Chaosvolt: Tempting, though adding a crafting recipe for a makeshift rolling press would be silly, if it's only used to make one item. Then again, maybe sheet metal could use it too.

File: data/json/recipes/food/veggy.json (hunk 2 above)
Comment (mugling) [inline data/json/recipes/food/veggy.json]: A corner case, but ugly no doubt. 

Arguably there's a lot of redundancy in that `recipe`. We need a `using: fruit` group or something of that nature. What groups have you identified so far?

**Comment by DangerNoodle** [comment]
Not certain about adding a makeshift rolling press. What other uses would it have for future content?

**Comment by Soadreqm** [comment]
>Also, the idea about crafting aluminum scrap from cans was brought up and dropped before because realism, as aluminum can't be smelted like that without oxidizing.

I think it can, actually. There are videos on Youtube of people melting down aluminum cans in makeshift crucibles. You presumably lose some of it to oxidization, but it might be worthwhile if you have a lot of empty aluminum cans and a need for aluminum ingots.
That said, there isn't a lot of aluminum in a can to begin with, so the cans/ingot ratio would be awful.
The other use for empty cans I can think of would be grinding them down into powder and using that as a chemical reagent.

>Not certain about adding a makeshift rolling press. What other uses would it have for future content?

Besides sheet metal, maybe some higher-level metal armor recipes? I'm not sure if it's worth it, but the concept of building industrial machinery out of salvaged cars sounds kinda cool.

**Comment by DangerNoodle** [comment]
Possibly. I did increase the ratio of cans to ingots. Or rather, ratio of cans to aluminum powder, and thus used the resulting ratios of accepted ingredients to hash out ratios for the ingot recipe. I did not increase it to what might be a realistic level, though. Enough that it might be impractical for large-scale demands, but acceptable for smaller requirements.

Some method of hammering aluminum flat might be better, but it might require altering charges and stack sizes of the foil, so that a single (post-resize) ingot of aluminum can make a smaller amount while still have the normal quantity spawn,

**Comment by DangerNoodle** [comment]
For now I am removing my addition of crafting aluminum foil, as it is not really as high a priority.

Should I add any alternatives to foil for baking pizzas and pies, like paper? Or remove that due to not being as useful as the "foil as an alternative to water" option added to baked vegetables?

**Comment by BorkBorkGoesTheCode** [comment]
IF you keep some form of aluminum foil as a recipe ingredient, could you add it as a byproduct?

**Comment by DangerNoodle** [comment]
That would entail adding more recipes for each such item, but I could do so if desired.

**Comment by Soadreqm** [comment]
What is the foil intended to do in the pizza and pie recipes?

**Comment by DangerNoodle** [comment]
If I recall, it was due to the fact we lack anything one could actually bake food with. However, that is a redundant detail.

And if that feature is cast aside, it likely is not needed for pizza, as it would not need a pan unless made deep-dish style. Foil as a makeshift pie tin maybe, but this also might be an unneeded detail.

**Comment by DangerNoodle** [comment]
Fixing merge conflict. Seems this branch was quite a bit out of date.

**Comment by DangerNoodle** [comment]
And it seems that doing so re-introduced a large number of lint errors. I am unsure how that occurred.

**Comment by mugling** [comment]
Unrelated to your linting errors but your rebase didn't work. You need to pull `master` before trying for a rebase.

**Comment by DangerNoodle** [comment]
I Thought I did so. Given the amount of work this caused, fixing the merge conflicts manually would have been simpler.

**Comment by mugling** [comment]
Did you identify any sets (fruit, veggies) etc for which we could implement a requirement group? This isn't a strict requirement for this PR but could be useful here

**Comment by DangerNoodle** [comment]
The only example that comes to mind would be the fruit ratios for fruit pies and pizzas. I can implement that, but first I am going to fix this by hand.

**Comment by DangerNoodle** [comment]
That should fix it. I should have gone with the tedious way to begin with.

**Comment by DangerNoodle** [comment]
For the time being, this should be ready.

File: data/json/recipes/food/veggy.json [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    "time": 18000,
+    "autolearn": true,
+    "byproducts": "plant_fibre",
+    "qualities": [
+      { "id": "CUT", "level": 1 },
+      { "id": "COOK", "level": 1 }
+    ],
+    "tools": [ [ [ "hotplate", 5 ], [ "char_smoker", 5 ], [ "toolset", 5 ], [ "fire", -1 ] ] ],
+    "components": [
+      [ [ "water", 1 ], [ "water_clean", 1 ], [ "aluminum_foil", 5 ] ],
```
Comment (Coolthulhu) [inline data/json/recipes/food/veggy.json]: Stalk originally didn't take water, suggesting it can be cooked over fire. If it is to take water/foil now, it should require COOK 2.

File: data/json/recipes/food/veggy.json [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_VEGGI",
+    "skill_used": "cooking",
+    "difficulty": 1,
+    "time": 15000,
+    "autolearn": true,
+    "qualities": [ { "id": "COOK", "level": 1 } ],
+    "tools": [ [ [ "hotplate", 3 ], [ "char_smoker", 3 ], [ "toolset", 3 ], [ "fire", -1 ] ] ],
+    "components": [
+      [ [ "water", 1 ], [ "water_clean", 1 ], [ "aluminum_foil", 3 ] ],
```
Comment (Coolthulhu) [inline data/json/recipes/food/veggy.json]: All foil recipes other than potato and shroom use 5 foil.

File: data/json/recipes/other/vehicle.json [hunk 5]
```diff
@@ -1,5 +1,19 @@
...
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_PARTS",
+    "skill_used": "fabrication",
+    "difficulty": 1,
+    "time": 8000,
+    "reversible": true,
+    "autolearn": true,
+    "using": [ [ "welding_standard", 6 ] ],
+    "qualities": [ { "id": "GLARE", "level": 2 } ],
+    "components": [ [ [ "material_aluminium_ingot", 16 ] ] ]
```
Comment (Coolthulhu) [inline data/json/recipes/other/vehicle.json]: Is xlframe material set to aluminum in the item definition?

File: data/json/recipes/other/materials.json [hunk 6]
```diff
@@ -32,5 +32,21 @@
...
+    "skill_used": "fabrication",
+    "difficulty": 0,
+    "time": 2500,
+    "autolearn": true,
+    "qualities": [
+      { "id": "HAMMER", "level": 1 },
+      { "id": "CONTAIN", "level": 1 }
+    ],
+    "tools": [ [ [ "toolset", 10 ], [ "fire", -1 ] ] ],
+    "components": [ [ [ "chem_aluminium_powder", 25 ], [ "can_drink_unsealed", 4 ], [ "aluminum_foil", 100 ] ] ]
```
Comment (Coolthulhu) [inline data/json/recipes/other/materials.json]: Not sure about this recipe. Melting 4 cans into a solid block of aluminum is weird.
It would also not produce vehicle frame grade metal.

I'll need good justification for this. Something better than "this will allow getting more aluminum" and "it will give an extra use to cans".

Comment (BorkBorkGoesTheCode) [inline data/json/recipes/other/materials.json]: The aluminum ingot weighs 2000 grams; a can weighs 13 grams.

Comment (DangerNoodle) [inline data/json/recipes/other/materials.json]: Weight is a wonky way to go about. Compare making aluminum powder, for example. A single 4-volume ingot (before the volume was reduced) was considered equivalent to 6 cans.

That said, I forgot to reduce the weight of ingots when I reduced their volume. So that should be 500 grams. I should reduce the weight to be consistent with the volume reduction before I work out a sane ratio.

File: data/json/recipes/food/veggy.json (hunk 4 above)
Comment (DangerNoodle) [inline data/json/recipes/food/veggy.json]: I had some using 7, I could have sworn. I was unsure of the desired ratio, so I initially went with equivalent to the amount of hotplate/toolset charges consumed.

File: data/json/recipes/food/veggy.json (hunk 3 above)
Comment (DangerNoodle) [inline data/json/recipes/food/veggy.json]: Can remove this requirement instead.

File: data/json/recipes/other/vehicle.json (hunk 5 above)
Comment (DangerNoodle) [inline data/json/recipes/other/vehicle.json]: Fixing now.

**Comment by DangerNoodle** [comment]
And merge conflict should be resolved for the second time, manually in this case. Any further issue with these changes? This hopefully should be ready.

**Comment by DangerNoodle** [comment]
This is still conflicting? Irritating.

**Comment by DangerNoodle** [comment]
I have examined the file and there should be no further cause for conflict in `recipe_food`, except for the effects of having ported over recipes to the desired files, and one instance of spacing corrections. If @mugling or others perhaps could see whether I am missing any regressions?

File: data/json/items/resources/metals.json [hunk 7]
```diff
@@ -138,5 +138,21 @@
...
+    "color": "light_gray",
+    "name": "aluminum ingot",
+    "category": "spare_parts",
+    "description": "A small aluminum ingot, standardized for further processing.  Light but durable, this could be cast into various shapes for construction or ground down to a powder, for more... high-profile applications.",
+    "price": 500,
+    "price_postapoc": 100,
+    "material": "aluminum",
+    "weight": 500,
+    "volume": 1,
+    "bashing": 1,
```
Comment (mugling) [inline data/json/items/resources/metals.json]: Doesn't need useless melee statistics

File: data/json/items/resources/metals.json [hunk 8]
```diff
@@ -138,5 +138,21 @@
     "color": "light_gray",
     "ammo_type": "components",
     "qualities": [ [ "HAMMER", 1 ] ]
+  },
+  {
+    "type":"GENERIC",
```
Comment (mugling) [inline data/json/items/resources/metals.json]: Not yet linted but the file is close to perfect so try and avoid inconsistent ordering here

File: data/json/items/resources/metals.json [hunk 9]
```diff
@@ -138,5 +138,21 @@
...
+    "id": "material_aluminium_ingot",
+    "symbol": ",",
+    "color": "light_gray",
+    "name": "aluminum ingot",
+    "category": "spare_parts",
+    "description": "A small aluminum ingot, standardized for further processing.  Light but durable, this could be cast into various shapes for construction or ground down to a powder, for more... high-profile applications.",
+    "price": 500,
+    "price_postapoc": 100,
+    "material": "aluminum",
+    "weight": 500,
```
Comment (mugling) [inline data/json/items/resources/metals.json]: Given a density of `2.7g/cm3` would give a weight of `675`. See examples in this file for how to document that

File: data/json/recipes/other/materials.json [hunk 10]
```diff
@@ -32,5 +32,21 @@
...
+    "skill_used": "fabrication",
+    "difficulty": 0,
+    "time": 2500,
+    "autolearn": true,
+    "qualities": [
+      { "id": "HAMMER", "level": 1 },
+      { "id": "CONTAIN", "level": 1 }
+    ],
+    "tools": [ [ [ "toolset", 10 ], [ "fire", -1 ] ] ],
+    "components": [ [ [ "chem_aluminium_powder", 25 ], [ "can_drink_unsealed", 6 ], [ "aluminum_foil", 100 ] ] ]
```
Comment (mugling) [inline data/json/recipes/other/materials.json]: A can weighs more than 100g? I'd drop this recipe for now whilst we work out balance

File: data/json/recipes/other/other.json [hunk 11]
```diff
@@ -10,5 +10,21 @@
...
+    "type": "recipe",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_OTHER",
+    "skill_used": "fabrication",
+    "difficulty": 5,
+    "time": 5000,
+    "autolearn": true,
+    "qualities": [ { "id": "SAW_M", "level": 1 } ],
+    "components": [
+      [ [ "material_aluminium_ingot", 4 ], [ "can_drink_unsealed", 16 ], [ "aluminum_foil", 400 ] ],
```
Comment (mugling) [inline data/json/recipes/other/other.json]: Same conservation of mass issue. This is important as otherwise we end up allowing crafting cycles that create infinite mass

File: data/json/recipes/other/tool.json [hunk 12]
```diff
@@ -90,6 +90,28 @@
     ]
   },
   {
+    "result": "pickaxe",
```
Comment (mugling) [inline data/json/recipes/other/tool.json]: Unrelated changes?

File: data/json/recipes/other/tool.json [hunk 13]
```diff
@@ -107,5 +129,22 @@
...
+    "difficulty": 3,
+    "time": 30000,
+    "autolearn": true,
+    "using": [ [ "forging_standard", 1 ] ],
+    "qualities": [ { "id": "HAMMER", "level": 2 } ],
+    "tools": [
+      [ [ "anvil", -1 ] ],
+      [ [ "forge", 150 ], [ "oxy_torch", 30 ] ]
+    ],
+    "components": [ [ [ "steel_lump", 1 ], [ "steel_chunk", 4 ], [ "scrap", 12 ], [ "pipe", 3 ], [ "material_aluminium_ingot", 4 ] ] ]
```
Comment (mugling) [inline data/json/recipes/other/tool.json]: Should be one ingot according to conservation of mass

File: data/json/recipes/other/tool.json [hunk 14]
```diff
@@ -107,5 +129,22 @@
...
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "fabrication",
+    "difficulty": 3,
+    "time": 30000,
+    "autolearn": true,
+    "using": [ [ "forging_standard", 1 ] ],
+    "qualities": [ { "id": "HAMMER", "level": 2 } ],
+    "tools": [
+      [ [ "anvil", -1 ] ],
+      [ [ "forge", 150 ], [ "oxy_torch", 30 ] ]
```
Comment (mugling) [inline data/json/recipes/other/tool.json]: `using forging_standard`

File: data/json/recipes/other/vehicle.json (hunk 5 above)
Comment (mugling) [inline data/json/recipes/other/vehicle.json]: Should only need ~8 ingots to conserve mass

File: data/json/recipes/food/carnivore.json [hunk 15]
```diff
@@ -0,0 +1,72 @@
+[
```
Comment (mugling) [inline data/json/recipes/food/carnivore.json]: What actually changed in this file?

File: data/json/recipes/food/veggy.json [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+    "subcategory": "CSC_FOOD_VEGGI",
+    "skill_used": "cooking",
+    "difficulty": 0,
+    "time": 15000,
+    "batch_time_factors": [ 67, 5 ],
+    "autolearn": true,
+    "qualities": [ { "id": "COOK", "level": 2 } ],
+    "tools": [ [ [ "hotplate", 7 ], [ "char_smoker", 7 ], [ "toolset", 7 ], [ "fire", -1 ] ] ],
+    "components": [
+      [ [ "water", 1 ], [ "water_clean", 1 ], [ "aluminum_foil", 5 ] ],
```
Comment (mugling) [inline data/json/recipes/food/veggy.json]: Aluminium foil or water is a reasonable mechanic. It would be good to abstract this is a `using` set but I'm not sure what you would name it?

File: data/json/recipes/food/veggy.json [hunk 17]
```diff
@@ -0,0 +1,219 @@
...
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_VEGGI",
+    "skill_used": "cooking",
+    "difficulty": 1,
+    "time": 15000,
+    "autolearn": true,
+    "qualities": [ { "id": "COOK", "level": 2 } ],
+    "tools": [ [ [ "hotplate", 3 ], [ "char_smoker", 3 ], [ "toolset", 3 ], [ "fire", -1 ] ] ],
+    "components": [
+      [ [ "water", 1 ], [ "water_clean", 1 ], [ "aluminum_foil", 5 ] ],
```
Comment (mugling) [inline data/json/recipes/food/veggy.json]: Baked potatoes don't involve water?

File: data/json/recipes/food/veggy.json [hunk 18]
```diff
@@ -0,0 +1,219 @@
...
+    "autolearn": true,
+    "qualities": [ { "id": "COOK", "level": 2 } ],
+    "tools": [ [ [ "hotplate", 5 ], [ "char_smoker", 5 ], [ "toolset", 5 ], [ "fire", -1 ] ] ],
+    "components": [
+      [ [ "water", 1 ], [ "water_clean", 1 ], [ "aluminum_foil", 5 ] ],
+      [ [ "veggy_wild", 1 ] ]
+    ]
+  },
+  {
+    "result": "cooked_cattail_stalk",
```
Comment (mugling) [inline data/json/recipes/food/veggy.json]: Unrelated changes?

File: data/json/recipes/other/other.json (hunk 11 above)
Comment (DangerNoodle) [inline data/json/recipes/other/other.json]: This is a pre-existing issue, as cans already could be broken down into powder.

File: data/json/recipes/other/tool.json (hunk 12 above)
Comment (DangerNoodle) [inline data/json/recipes/other/tool.json]: Odd, uncertain why the pickaxe change is in this pull request.

Comment (mugling) [inline data/json/recipes/other/tool.json]: You need to break this up. Start with the ingots as that's simplest.

File: data/json/recipes/food/veggy.json (hunk 18 above)
Comment (DangerNoodle) [inline data/json/recipes/food/veggy.json]: This was ported over from the recipe_food file, and originally changed to allow foil before Coolthulho advised against it.

File: data/json/recipes/food/carnivore.json (hunk 15 above)
Comment (DangerNoodle) [inline data/json/recipes/food/carnivore.json]: Pizza and pie items were ported over and originally required foil, before being dropped as not sensible.

**Comment by DangerNoodle** [comment]
Closing this to make a new pull request for the ingot changes and related aluminum uses. Cooking with foil might be interesting but is not needed for this, nor the mining mod.

**Comment by mugling** [comment]
Cooking with foil is an interesting idea and could be more relevant if we end up with desert biomes. I wouldn't give up on it.

**Comment by DangerNoodle** [comment]
A separate pull request, possibly. I will work on it later, possibly. Losing the energy to handle so many changes at one time.
